nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/ConsumeGCPubSubLite.java

```diff
@@ -0,0 +1,258 @@
+        final Collection<ValidationResult> results = new ArrayList<ValidationResult>(1);
+        final String subscription = validationContext.getProperty(SUBSCRIPTION).evaluateAttributeExpressions().getValue();
+
+        try {
+            SubscriptionPath.parse(subscription);
+        } catch (ApiException e) {
+            results.add(new ValidationResult.Builder()
+                    .subject(SUBSCRIPTION.getName())
+                    .input(subscription)
+                    .valid(false)
+                    .explanation("The Suscription does not have a valid format.")
+                    .build());
+        }
+
+        return results;
+    }
+
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        try {
+            subscriber = getSubscriber(context);
+        } catch (Exception e) {
+            getLogger().error("Failed to create Google Cloud Subscriber", e);
+            throw new ProcessException(e);
+        }
+        try {
+            subscriber.startAsync().awaitRunning();
+        } catch (Exception e) {
+            getLogger().error("Failed to create Google Cloud Subscriber", subscriber.failureCause());
+            throw new ProcessException(e);
+        }
+    }
+
+    @OnStopped
+    public void onStopped() {
+        try {
+            if (subscriber != null) {
+                subscriber.stopAsync().awaitTerminated();
+            }
+        } catch (Exception e) {
+            getLogger().warn("Failed to gracefully shutdown the Google Cloud PubSub Lite Subscriber", e);
+        }
+    }
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return ImmutableList.of(SUBSCRIPTION,
+                GCP_CREDENTIALS_PROVIDER_SERVICE,
+                BYTES_OUTSTANDING,
+                MESSAGES_OUTSTANDING);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return Collections.singleton(REL_SUCCESS);
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        if (subscriber == null) {
+            getLogger().error("Google Cloud PubSub Subscriber was not properly created. Yielding the processor...");
+            context.yield();
+            return;
+        }
+
+        if(!subscriber.isRunning()) {
+            getLogger().error("Google Cloud PubSub Subscriber is not running. Yielding the processor...", subscriber.failureCause());
+            throw new ProcessException(subscriber.failureCause());
+        }
+
+        Message message = messages.poll();
+        if(message == null) {
+            context.yield();
+            return;
+        }
+
+        FlowFile flowFile = session.create();
+
+        final Map<String, String> attributes = new HashMap<>();
+        attributes.put(MESSAGE_ID_ATTRIBUTE, message.getMessage().getMessageId());
+        attributes.put(ORDERING_KEY_ATTRIBUTE, message.getMessage().getOrderingKey());

```

Comment:
Just to confirm, this would only be populated if someone set the ordering key by publishing external to NiFi?  I couldn't find a way to set the ordering key on the `PublishGCPubSubLite` processor, but did see it being set when publishing using `gcloud`.  Perhaps a feature for another day.